test/scalability/checker/checker_test.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package checker
+
+import (
+	"flag"
+	"os"
+	"testing"
+
+	"sigs.k8s.io/yaml"
+
+	"sigs.k8s.io/kueue/test/scalability/runner/recorder"
+	"sigs.k8s.io/kueue/test/scalability/runner/stats"
+)
+
+var (
+	summaryFile  = flag.String("summary", "", "the runner summary report")
+	cmdStatsFile = flag.String("cmdStats", "", "command stats yaml file")
+	rangeFile    = flag.String("range", "", "expectations range file")
+)
+
+type RangeSpec struct {
+	Cmd struct {
+		MaxMSWall int64
+		MaxMSUser int64
+		MaxMSSys  int64
+		Maxrss    uint64
+	}
+	ClusterQueueClassesMinUsage  map[string]float64
+	WlClassesMaxAvgMSToAdmission map[string]int64
```

Comment:
Since it's done, let's keep it.

Reply:
@trasc we all strive to run maintainable code.

The fact that the code is already written doesn't imply that it cannot be changed. Please consider the suggestion seriously.

Reply:
The problem is that the proposed way, using 1s scraped metrics, in some not yet defined format / way to process does not have the necessary precision.

@alculquicondor, @mimowo   The implemented way was discussed with both of you offline and agreed upon two weeks ago .

Reply:
Tbh, I was sceptical about the approach during the meeting, but didn't raise it because this was early on and I didn't have full context like seeing the code. Now as I reviewed the code and played with the tool myself, I think this is an unnecessary complication.

Regarding the scraping of 1s interval this is much more than used for most kubernetes scalability tests, which is 15s by default.

Regarding the format, yes this is a weak point, but it is a weak point of either solution. We will dump the metrics to csc as a time series, it can be easily visualized. We could even generate the image as part of the summary.

Reply:
Iirc the meeting was 30min and we didn't discuss this subject longer than 5min, so I wouldn' t assume this was a 'written in stone' decision.Also iirc, Also raised a question about metrics. finally, I spent time discussing this with Marcin later on too.

Reply:
"the meeting was 30min and we didn't discuss this subject longer than 5min" until Aldo left and another 30+ just me and you.

Also that 5min ware more.

Reply:
IIRC, the rationale from your side back then was that using metrics would require a significant amount of code. Based on that assumption, we gave an ok to explore this path.

However, now this approach seems to have a lot of code as well, while being farther from standard practices for monitoring (that is, the use of metrics).

I think the use of metrics should be seriously considered, even if it means rewriting some parts of this PR.

Reply:
https://github.com/kubernetes-sigs/kueue/pull/1931#issuecomment-2049414703